Include/internal/pycore_object.h

```diff
@@ -159,6 +159,24 @@ static inline void _Py_RefcntAdd(PyObject* op, Py_ssize_t n)
 }
 #define _Py_RefcntAdd(op, n) _Py_RefcntAdd(_PyObject_CAST(op), n)
 
+static inline int _Py_Reuse_Immutable_Object(PyObject *ob) {
+    /* Function to check whether an immutable object such as a tuple can be
+     * modified inplace by the owning method
+     */
+#if !defined(Py_GIL_DISABLED)
+    return Py_REFCNT(ob) == 1;
+#else
+    uint32_t local = _Py_atomic_load_uint32_relaxed(&ob->ob_ref_local);
+    if (local != 1)
+        return 0;
+    Py_ssize_t shared = _Py_atomic_load_ssize_relaxed(&ob->ob_ref_shared);
+    if (shared != 0) {
+        return 0;
+    }
+    return ob->ob_tid == _Py_ThreadId();
```

Comment:
Let's restructure this so that the thread id check is first. I don't think it matters in this case, but it more closely follows the biased reference counting logic:

```suggestion
    // NOTE: the entire ob_ref_shared field must be zero, including flags, to
    // ensure that other threads cannot concurrently create new references to
    // this object. 
    return (_Py_IsOwnedByCurrentThread(ob) &&
            _Py_atomic_load_uint32_relaxed(&ob->ob_ref_local) == 1 &&
            _Py_atomic_load_ssize_relaxed(&ob->ob_ref_shared) == 0);
```